File: src/DistributedFlowLab.Domain/Enums/NodeType.cs
namespace DistributedFlowLab.Domain.Enums;

/// <summary>
/// Canonical vocabulary of node types a <see cref="Entities.Scenario"/> can contain.
/// Mirrored verbatim by the frontend (`web/src/domain/nodeType.ts`).
/// See .docs/02-architecture/data-model.md §3.1.
/// </summary>
public enum NodeType
{
    Producer,
    Consumer,
    Service,
    ApiGateway,
    LoadBalancer,
    Exchange,
    Queue,
    Topic,
    Partition,
    Broker,
    Database,
    Cache,
    DeadLetterQueue,
    Client,
}

File: web/src/domain/nodeType.ts
/**
 * Canonical NodeType vocabulary — mirrors DistributedFlowLab.Domain `NodeType`
 * (.docs/02-architecture/data-model.md §3.1). Keep in exact sync with the backend enum.
 */
export const NODE_TYPES = [
  'Producer',
  'Consumer',
  'Service',
  'ApiGateway',
  'LoadBalancer',
  'Exchange',
  'Queue',
  'Topic',
  'Partition',
  'Broker',
  'Database',
  'Cache',
  'DeadLetterQueue',
  'Client',
] as const

export type NodeType = (typeof NODE_TYPES)[number]
